homeassistant/components/sonos/media_player.py

```diff
@@ -533,12 +546,19 @@ def _set_favorites(self):
                 # Skip unknown types
                 _LOGGER.error("Unhandled favorite '%s': %s", fav.title, ex)
 
+    def _set_queue(self):
+        """Set available queue."""
+        self._queue = []
+        for item in self.soco.get_queue():
+            self._queue.append(item.title)
```

Comment:
I don't think you need to copy the queue, you can probably just do: `self._queue = self.soco.get_queue()`.

Reply:
```
2020-05-25 08:41:03 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection.4543351056] Unable to serialize to JSON. Bad data found at $.result[2](state: media_player.living_surround).attributes.queue[0]=<DidlMusicTrack 'b'Graspop 2019, Friday June 21st'' at 0x111dc0f50>(<class 'pysonos.data_structures.DidlMusicTrack'>, $.result[5 ...............
```

Reply:
Sorry, I was too fast and didn't notice that you are appending just the title, not each item.

So it could be like this instead: `self._queue = [item.title for item in self.soco.get_queue()]` but that is less of an obvious improvement :)

Reply:
Can you point me in the right direction for this

> grouped players show their own, inactive queue information while I think they should show the group's master information

Reply:
Make a property like `media_title` that has the `@soco_coordinator` decorator. This will automatically forward the call to the coordinator when the entity is a group slave.

Then use this new property in `device_state_attributes`.